### DISCUSSION TITLE
Double-Definition Dungeon

### DISCUSSION THREAD

**Discussion by OCRenkist** [jeremyshannon/Exile#65] [Q&A]
I’m stuck on parsing the penultimate biome from the original set, whereas it has dual declarations of both `node_dungeon` and `node_dungeon_stair` — does the latter parameter pair abrogate the former? If they were to be blended, I should expect to see the use of a list in curly braces, within each of a single declaration. It’s quite a mystery. 🤔

```
mods/mapgen/biomes.lua
...
374: name               = "underground",
...
376: node_dungeon       = "nodes_nature:basalt_brick",
377: node_dungeon_stair = "stairs:stair_basalt_block",
...
380: node_dungeon       = "nodes_nature:granite_brick",
381: node_dungeon_stair = "stairs:stair_granite_block",
```

**Comment by jeremyshannon** [comment]
I think the latter overwrites the former? Line ~7905 of lua_api.txt is unclear about whether the example above it defines one node, then overwrites it with two, or if it's defining three options (66% water, 33% lava). I would hazard a guess that overwriting makes more sense and follows the law of least surprise.
